frontend/src/pages/docs/[...slug].tsx

```diff
@@ -1,75 +1,158 @@
-// }
-
-// const BlogPage = ({ mdxSource }: Props) => {
-//   return (
-//     <>
-//       <MDXRemote {...mdxSource} components={components} />
-//     </>
-//   );
-// };
-
-function BlogPage(): JSX.Element {
-  return <div />;
+import { Link } from "czifui";
+import fs from "fs";
+import matter from "gray-matter";
+import { GetStaticPaths } from "next";
+import { MDXRemote, MDXRemoteSerializeResult } from "next-mdx-remote";
+import { serialize } from "next-mdx-remote/serialize";
+import Image from "next/image";
+import NextLink from "next/link";
+import pathTool from "path";
+import styled from "styled-components";
+
+const DOC_SITE_FOLDER_NAME = "doc-site";
+
+interface Directory {
+  dirName: string;
+  files: Array<string>;
+  slug: Array<string>;
+  subDirectories: Array<Directory>;
+}
+
+const CACHED_FILE_PATHS = new Map<Array<string>, Directory>();
+function filePaths(...root: Array<string>): Directory {
+  const cacheStore = CACHED_FILE_PATHS && CACHED_FILE_PATHS.get(root);
+  if (cacheStore) {
+    console.log("CACHE HIT");
+    return cacheStore;
+  }
+
+  const newDirectory = {
+    dirName: root[root.length - 1] ?? "",
+    files: [],
+    slug: root,
+    subDirectories: [],
+  } as Directory;
+
+  const paths = fs.readdirSync(pathTool.join(DOC_SITE_FOLDER_NAME, ...root));
+  paths.forEach((path) => {
+    const builtFullPath = pathTool.join(DOC_SITE_FOLDER_NAME, ...root, path);
+    if (fs.lstatSync(builtFullPath).isFile()) {
+      if (path.endsWith(".mdx"))
+        newDirectory.files.push(path.replace(".mdx", ""));
+    } else {
+      newDirectory.subDirectories.push(filePaths(...root, path));
+    }
+  });
+  CACHED_FILE_PATHS.set(root, newDirectory);
+  return newDirectory;
+}
+
+function generatePaths(
+  directory: Directory,
+  slugs = new Array<{ params: { slug: Array<string> } }>()
+): Array<{ params: { slug: Array<string> } }> {
+  directory.files.forEach((file) => {
+    slugs.push({ params: { slug: [...directory.slug, file] } });
+  });
+  if (directory.subDirectories.length > 0) {
+    directory.subDirectories.forEach((subDirectory) => {
+      slugs.push(...generatePaths(subDirectory, slugs));
+    });
+  }
+  return slugs;
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const filePath = filePaths();
+  const paths = generatePaths(filePath);
+
+  return {
+    fallback: false,
+    paths,
+  };
+};
+
+export const getStaticProps = async ({
+  params: { slug },
+}: {
+  params: { slug: Array<string> };
+}): Promise<{
+  props: Props;
+}> => {
+  const markdownWithMeta = fs.readFileSync(
+    pathTool.join("doc-site", slug.join("/") + ".mdx"),
+    "utf-8"
+  );
+
+  const filePath = filePaths();
+  const { data: frontMatter, content } = matter(markdownWithMeta);
+  const mdxSource = await serialize(content);
+  return {
+    props: {
+      filePath,
+      frontMatter,
+      mdxSource,
+      slug,
+    },
+  };
+};
+
+const RenderDirectory = ({ directory }: { directory: Directory }) => {
```

Comment:
nit: Since this is a component, maybe just `Directory`?

Reply:
you're right!